### DISCUSSION TITLE
How to upgrade many Talos nodes automatically ?

### DISCUSSION THREAD

**Discussion by rancherr** [siderolabs/talos#9105] [General]
Hei,
I would like to upgrade many Talos nodes automatically... is there best practice option?
There is many tips out there how to upgrade talos... but how to automate it?
Ansible - no talos module - you need to write the script
OpenTofu? - no talos module

Any ideas for opensource solution?

**Comment by smira** [comment]
From Talos point of view, upgrades are an API call, so you can build the automation that feels right to you.

Make sure you check `talosctl health` between the upgrades, and do any additional checks you might need for your workload stuff.

[Omni](https://github.com/siderolabs/omni) which is source available, provides automated upgrades.
